src/processors/BaseProcessor.cpp

```diff
@@ -4,12 +4,16 @@
 
 BaseProcessor::BaseProcessor (const String& name,
                               ParamLayout params,
-                              UndoManager* um,
-                              int nInputs,
-                              int nOutputs) : JuceProcWrapper (name),
-                                              vts (*this, um, Identifier ("Parameters"), std::move (params)),
-                                              numInputs (nInputs),
-                                              numOutputs (nOutputs)
+                              UndoManager* um) : BaseProcessor (
+    name,
+    std::move (params),
+    BasicInputPort {},
+    BasicOutputPort {},
+    um,
+    [] (auto)
+    { return PortType::audio; },
+    [] (auto)
+    { return PortType::audio; })
 {
     onOffParam = vts.getRawParameterValue ("on_off");
```

Comment:
Since this constructor already delegates to the other constructor, we no longer need to do anything in the constructor body.

Reply:
yes, good point!